ui/admin/app/controllers/scopes/scope/roles/role/scopes.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Controller, { inject as controller } from '@ember/controller';
+import {
+  GRANT_SCOPE_THIS,
+  GRANT_SCOPE_CHILDREN,
+  GRANT_SCOPE_DESCENDANTS,
+} from 'api/models/role';
+
+export default class ScopesScopeRolesRoleScopesController extends Controller {
+  @controller('scopes/scope/roles/index') roles;
+
+  isKeywordThis(id) {
```

Comment:
Do we need separate methods? Since we are passing in the ID from the template, can we just check if the ID exists in the array of grants such as [GRANT_SCOPE_THIS, GRANT_SCOPE_CHILDREN,
  GRANT_SCOPE_DESCENDANTS]?

Reply:
Actually looking this over I think I can refactor this to one specific for this as we want to use the current scope of the role if they have "this". Along with a second method that checks if it is "children" or "descendants" and just use the id in my translations to get the correct text. I will refactor this now. Thanks!

Reply:
Done.